packages/instrumentation-llamaindex/src/custom-llm-instrumentation.ts

```diff
@@ -0,0 +1,175 @@
+import * as lodash from "lodash";
+import type * as llamaindex from "llamaindex";
+
+import { Tracer, SpanStatusCode, trace, context } from "@opentelemetry/api";
+import { safeExecuteInTheMiddle } from "@opentelemetry/instrumentation";
+
+import { SpanAttributes } from "@traceloop/ai-semantic-conventions";
+
+import { LlamaIndexInstrumentationConfig } from "./types";
+import { shouldSendPrompts } from "./utils";
+
+export class CustomLLMInstrumentation {
+  private config: LlamaIndexInstrumentationConfig;
+  private tracer: Tracer;
+
+  constructor(config: LlamaIndexInstrumentationConfig, tracer: Tracer) {
+    this.config = config;
+    this.tracer = tracer;
+  }
+
+  completionWrapper({ className }: { className: string }) {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const plugin = this;
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    return (original: Function) => {
+      return function method(this: llamaindex.LLM, ...args: unknown[]) {
+        const prompt = args[0];
+
+        const span = plugin.tracer.startSpan(
+          `${lodash.snakeCase(className)}.completion`,
+        );
+
+        span.setAttribute(SpanAttributes.LLM_VENDOR, "llamaindex");
```

Comment:
Is there a way to figure out the actual llm vendor here? Cause I'd say that if we're using OpenAI for example than writing OpenAI here is more correct

Reply:
I can replace it with class name of LLM. That's I think the closest thing to vendor. We've mapped LLM_VENDOR to class name in python instrumentation.